packages/meteor/dynamics_browser.js

```diff
@@ -36,11 +36,23 @@ EVp.getOrNullIfOutsideFiber = function () {
  */
 EVp.withValue = function (value, func) {
   var saved = currentValues[this.slot];
+
   try {
     currentValues[this.slot] = value;
+
     var ret = func();
+    var isPromise = Meteor._isPromise(ret);
+
+    if (isPromise) {
+      var self = this;
+      return ret.finally(function () {
+        currentValues[self.slot] = saved;
+      });
+    }
   } finally {
```

Comment:
The `meteor` package should support browsers that don't support optional catch binding. I'm surprised the check legacy syntax script doesn't catch this.